Firestore/Source/Public/FirebaseFirestore/FIRVectorValue.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(VectorValue)
+@interface FIRVectorValue : NSObject
```

Comment:
I payed around with this locally and I think it will make a more idiomatic Swift API to avoid the `toArray()` method in favor of a computed property, especially since the Codable conformance sort of forces us down that path. IIUC, this type is a kind of box/container type so I feel a property still makes sense to access the underlying data type.

I'll make code suggestions below.

Reply:
Thanks for the code suggestions. I updated this code to use the read-only `array` property as you suggested.